src/wp-includes/block-editor.php

```diff
@@ -665,6 +665,8 @@ function get_block_editor_settings( array $custom_settings, $block_editor_contex
 		}
 	}
 
+	$editor_settings['canUpdateBlockBindings'] = current_user_can( 'edit_block_binding', $block_editor_context->post->ID );
```

Comment:
This would need a safety check in case `$block_editor_context->post` doesn't exist.

Reply:
I don't think it's going to be reliable to use post ID. It will only work correctly for the existing post in the post editor. It is impossible to correctly detect the post id for the site editor as everything happens dynamically on the client.

Reply:
Yes, it seemed complicated when I was looking through the code and use cases. That's why, if we want to check against a post ID, I was adding a fallback to `edit_theme_options` capability: https://github.com/WordPress/wordpress-develop/pull/7258/files#diff-7b99cb99a8105f5254c9c282a8cb9eed9d9b48f2d974544ce6919fe124bd0427R813

That way, we don't try to detect the post id, if it isn't provided in the context and we are in the site editor, we map the `edit_block_binding` capability to `edit_theme_options`.

Having said that, maybe it is just easier to not pass a post ID and map it directly to other capabilities.

Reply:
> This would need a safety check in case $block_editor_context->post doesn't exist.

Confirmed:
https://github.com/WordPress/wordpress-develop/blob/7f4fd30dffdfe8d7d817c2cdb4ab98408f1c0add/src/wp-includes/class-wp-block-editor-context.php#L39

Reply:
This should be solved in this commit: https://github.com/WordPress/wordpress-develop/pull/7258/commits/a6a8d21393118d53ce91c636c9676613042fc9b8